### PR TITLE
Fix broken links in documentation

### LINKED ISSUE #6294: Broken links in docs
### What happened?

1.https://docs.lakefs.io/howto/deploy/aws.html the"DynamoDB! See the configuration reference for more information." configuration reference link is broken

### Expected Behavior

_No response_

### lakeFS Version

0.105

### Deplyoment

_No response_

### Affected Clients

_No response_

### Relevant logs output

_No response_

### Contact Details

_No response_

### PR DESCRIPTION
**Pull request by arielshaqed**
Also make them _relative_, which relies less on Jekyll-time magic and
therefore **easier to check** while working locally.

Fixes #6294.

### CODE REVIEW

**Comment by arielshaqed** [comment]
# Hiyush reviewers...

## How to review

Sorry for the too-large PR.  Unfortunately #6294 is the tip of the iceberg (and I don't mean the table format).  Many of our links were broken, for multiple reasons.  The main change is to [change the instructions for how to link](https://github.com/treeverse/lakeFS/pull/6305/files#diff-298f74a20864fdbaad2a48ce04b13e8f95af621145fde231f502895a437a79e4) -- please start by reviewing that one!  After that, many links were simply broken -- not only the ones referenced in the issue.  I don't know why our "check links" check failed to find them; the fact of the matter is that it did not.  Additionally, many of our _anchored_ links were also broken; I fixed all of those that I could find.

## What to review

Most of the changes follow one of these three mechanics:

* `{{ site.baseurl }}/path/to/page.html` ==> `{% link path/to/page.md %}`.
* But... `{% link ... %}` doesn't do redirects, it checks the link at the Jekyll level, so `{{ site.baseurl }}/path/to/directory/` ==> `{% link path/to/directory/index.md %}`.
* And also... there is no point in linking to `.html` when the page is called `.md`.  This is _not_ a redirect, Jekyll will generate a page with the extension `.html`.  But its link-checking still needs the correct name.
* I lied, there are actually four things: `{% link ... %}` doesn't work with anchors because Jekyll and moustache syntax something.  The workaround that I found was to use a link at the bottom of the text - that lets Jekyll do its thing and only _then_ Markdown does its thing for the link, so anchors suddenly work.

I did _not_ fix any link names in (non-Markdown...) assets.  There is no risk of breakage there: no anchors, no extensions change.  I believe that changing these would _increase_ risk of breakage.

Some of the changes reflect broken links that the Jekyll `{% link ... %}` was able to find.

## Why bother

The existing docs contain multiple breakages, and it seems that the Surge-based link checking was unable to find them.  It _does_ find other breakages, that Jekyll does not find, so best to keep **both**.  `{% link ... %}` is syntactic and can check some links, `{{site.baseurl}}` is a macro and prevents Jekyll from checking.  As always, the sooner we can check links the better.

File: .github/workflows/docs-pr.yaml [hunk 1]
```diff
@@ -6,6 +6,9 @@ on:
       - "docs/**"
     branches:
       - master
+  push:
+    paths:
+      - "docs/**"
```
Comment (johnnyaug) [inline .github/workflows/docs-pr.yaml:11]: Do we want this?

**Review by johnnyaug — APPROVED** [review]
Marvelous!

File: .github/workflows/docs-pr.yaml (hunk 1 above)
Comment (arielshaqed) [reply .github/workflows/docs-pr.yaml:11]: I think we do.  So I'll move it to another PR and ask you and @nopcoder to TAL.

**Review by arielshaqed — COMMENTED** [review]
!

File: .github/workflows/docs-pr.yaml (hunk 1 above)
Comment (arielshaqed) [reply .github/workflows/docs-pr.yaml]: Removed this, opened #6311.

**Comment by arielshaqed** [comment]
THANKS!

**Pull request merged**